desk/lib/dm.hoon

```diff
@@ -384,6 +406,32 @@
     =/  entry   (get ship `@da`time)
     ?~  entry  ``chat-paged-writs+!>(*paged-writs:c)
     (get-around mode.pole ver time.u.entry count)
+  ::
+      [%range start=@ end=@ mode=?(%light %heavy) ~]
+    ::TODO  support @da format in path for id (or timestamp) ranges?
+    =/  start=@ud
+      ?:  =(%$ start.pole)  1
+      (slav %ud start.pole)
+    =/  end=@ud
+      ?:  =(%$ end.pole)  num.pac
+      (slav %ud end.pole)
+    %-  give-paged-writs
+    :+  mode.pole
+      ver
+    ::  queries near end more common, so we make a newest-first list,
+    ::  and walk it "backwards" until we extract our desired range
+    ::
+    =/  wits=(list [time p=writ:c])
+      (bap:on:writs:c wit.pac)
+    =|  out=(list [time writ:c])
+    |-
+    ?~  wits  ~
+    ?:  (gth seq.p.i.wits end)
+      $(wits t.wits)
+    ?:  (lth seq.p.i.wits start)
+      ~  ::  done
+    :-  i.wits
+    $(wits t.wits)
```

Comment:
This is not tail-recursive, is it? We probably want to use `out` from above.

Reply:
What do you mean, not tail-recursive? We put the recursion in the tail of the noun we produce, it's the textbook case lol. You're correct that we don't use out (vestigial from an earlier implementation), we don't need to accumulate when we can produce the product noun "in place".

Reply:
I was thinking about the cons there and wasn't sure if the Hoon compiler knows how to optimize it. If it does that is good to know!